internal/store/util.go

```diff
@@ -32,3 +40,32 @@ func sliceToMap(inc []string) map[string]struct{} {
 
 	return m
 }
+
+func filterMap[V any](m map[string]V, keys []string, mode filterMode) map[string]V {
```

Comment:
This doesn't strictly need generics right?

Reply:
Not necessarily, but in this context I said why not :)